lib/tests/streamlit/runtime/caching/common_cache_test.py

```diff
@@ -575,3 +575,83 @@ def bar():
         # have been called just once.
         self.assertEqual(2, foo_call_count[0])
         self.assertEqual(1, bar_call_count[0])
+
+    @parameterized.expand(
+        [
+            ("memo", memo),
+            ("singleton", singleton),
+        ]
+    )
+    def test_without_spinner(self, _, cache_decorator):
+        """If the show_spinner flag is not set, the report queue should be
+        empty.
+        """
+
+        @cache_decorator(show_spinner=False)
+        def function_without_spinner(x: int) -> int:
+            # Without the timeout, the spinner is sometimes not available
+            time.sleep(0.2)
+            return x
+
+        function_without_spinner(3)
+        self.assertTrue(self.forward_msg_queue.is_empty())
+
+    @parameterized.expand(
+        [
+            ("memo", memo),
+            ("singleton", singleton),
+        ]
+    )
+    def test_with_spinner(self, _, cache_decorator):
+        """If the show_spinner flag is set, there should be one element in the
+        report queue.
+        """
+
+        @cache_decorator(show_spinner=True)
+        def function_with_spinner(x: int) -> int:
+            # Without the timeout, the spinner is sometimes not available
+            time.sleep(0.2)
+            return x
+
+        function_with_spinner(3)
+        self.assertFalse(self.forward_msg_queue.is_empty())
```

Comment:
nit: Maybe we check here if the spinner text contains `Running`

Reply:
Yes, I tried to write that test case too:
Something similar to:

```

    @parameterized.expand(
        [
            ("memo", memo),
            ("singleton", singleton),
        ]
    )
    def test_with_spinner(self, _, cache_decorator):
        """If the show_spinner flag is set, there should be one element in the
        report queue.
        """

        @cache_decorator(show_spinner="CUSTOM_TEXT")
        def function_with_custom_spinner_text(x: int) -> int:
            # Without the timeout, the spinner is sometimes not available
            time.sleep(0.2)
            return x

        function_with_custom_spinner_text(3)

        time.sleep(0.3)
        el = self.get_delta_from_queue().new_element
        self.assertEqual(el.spinner.text, "CUSTOM_TEXT")

```

Unfortunately, this doesn't work. And only `empty` element returned from queue.
 Look's like it works only when `time.sleep` invoked in with `st.spinner` context manager itself.
 https://github.com/streamlit/streamlit/blob/807598d3f8c0ccaaf9dee63836fee0d77802d11f/lib/tests/streamlit/streamlit_test.py#L465
 
 https://github.com/streamlit/streamlit/blob/807598d3f8c0ccaaf9dee63836fee0d77802d11f/lib/tests/streamlit/write_test.py#L240
 
I will remove all unnecessary `time.sleep` in current tests

Reply:
Ahh ok, I see. In that case it is fine as is :)